### PR TITLE
fix: migration to AuditBuilder

### PR DESCRIPTION
**Pull request by raji-unni**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/experimentation/handler.js [hunk 1]
```diff
@@ -42,55 +28,49 @@ function processRUMResponse(data) {
...
+  log.info(`Experimentation Audit completed in ${formattedElapsed} seconds for ${baseURL}`);
+  return auditData;
 }
+
+/**
+ * Creates an Experimentation audit runner function.
+ *
+ * @returns {function(*, *): Promise<Object>} - Returns the Experimentation audit runner function.
+ */
+export function createExperimentationAuditRunner() {
```
Comment (solaris007) [inline src/experimentation/handler.js:69]: if you don't need to support multiple experimentation handlers, you can drop the factory function and use the `experimentationAuditRunner` function directly with `.withRunner`.

Comment (raji-unni) [reply src/experimentation/handler.js]: Have incorporated these changes.

File: src/experimentation/handler.js [hunk 2]
```diff
@@ -42,55 +28,40 @@ function processRUMResponse(data) {
...
+  const endTime = process.hrtime(startTime);
+  const elapsedSeconds = endTime[0] + endTime[1] / 1e9;
+  const formattedElapsed = elapsedSeconds.toFixed(2);
+
+  log.info(`Experimentation Audit completed in ${formattedElapsed} seconds for ${baseURL}`);
+  return auditData;
 }
+
+export default new AuditBuilder()
+  .withRunner((finalURL, context) => experimentationAuditRunner(finalURL, context))
```
Comment (solaris007) [inline src/experimentation/handler.js:65]: you don't even need an anonymous function, you can do directly:
```suggestion
  .withRunner(experimentationAuditRunner)
```

Comment (raji-unni) [reply src/experimentation/handler.js]: Have implemented this.

**Review by solaris007 — APPROVED** [review]
thanks for your work :)

**Pull request merged**